platform/mellanox/mlnx-platform-api/sonic_platform/component.py

```diff
@@ -532,11 +538,12 @@ def auto_update_firmware(self, image_path, boot_action):
             return FW_AUTO_ERR_UKNOWN
 
         if boot_action in supported_boot:
-            if self.update_firmware(image_path):
-                # Successful update
-                return FW_AUTO_INSTALLED
-            # Failed update (unknown reason)
-            return FW_AUTO_ERR_UKNOWN
+            if boot_action == 'cold':
```

Comment:
@alexrallen suggest moving supported `boot_action` validation on top of this method

Reply:
Can you please clarify this suggestion? I am not sure what you mean.

Reply:
@alexrallen the validation of `boot_action` can be done on top in this method: if it doesn't fit the criteria - return immediately. In general it helps to avoid a nested if-else inside